### ISSUE TITLE
Missing dependency Az.Resources for Update-M365DSCAzureAdApplication

### ISSUE THREAD

**Issue by chaospheremk** [microsoft/Microsoft365DSC#2567]
#### Details of the scenario you tried and the problem that is occurring
Followed the user guide to install Microsoft365DSC by running:

```
Install-Module Microsoft365DSC -Force
Update-M365DSCDependencies
```

Ran the following with the intention of getting all update permissions and then creating an Azure AD App Registration for certificate authentication using Update-M365DSCAzureAdApplication:

```
$AllPermissions = Get-M365DSCCompiledPermissionList -ResourceNameList (Get-M365DSCAllResources)
$UpdatePermissions = $AllPermissions.UpdatePermissions

$PermList = @()
foreach ($UpdatePermission in $UpdatePermissions) {
    $PermObject = @{Api="Graph";PermissionName="$UpdatePermission"}
    $PermList += $PermObject
}

Connect-MgGraph -Scopes $UpdatePermissions

Update-M365DSCAzureAdApplication -ApplicationName 'Microsoft365DSC' -Permissions $PermList -AdminConsent -Type Certificate -CreateSelfSignedCertificate -CertificatePath c:\temp\M365DSC.cer
```
Then I am met with:

![image](https://user-images.githubusercontent.com/75006594/202959557-7371e39d-b0a4-4635-a88b-04a4102873ef.png)

#### Verbose logs showing the problem
N/A
#### Suggested solution to the issue
N/A
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
N/A

#### The operating system the target node is running
OsName               : Microsoft Windows 10 Enterprise
OsOperatingSystemSKU : EnterpriseEdition
OsArchitecture       : 64-bit
WindowsVersion       : 2009
WindowsBuildLabEx    : 19041.1.amd64fre.vb_release.191206-1406
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of the DSC module that was used ('dev' if using current dev branch)
Release 1.22.1109.1

**Comment by chaospheremk** [comment]
To add on to this, after manually installing Az.Resources with:
`Install-Module Az.Resources -Force`

And then trying to run Update-M365DSCAzureAdApplication:
`Update-M365DSCAzureAdApplication -ApplicationName 'Microsoft365DSC' -Permissions $PermList -AdminConsent -Type Certificate -CreateSelfSignedCertificate -CertificatePath c:\temp\M365DSC.cer`

I get the following:
![image](https://user-images.githubusercontent.com/75006594/202961879-f241cefa-bc43-4f65-94f5-e911b004f930.png)
![image](https://user-images.githubusercontent.com/75006594/202961959-e460a67b-4613-4536-a5e5-8b757e75d9bb.png)
![image](https://user-images.githubusercontent.com/75006594/202962042-cd234673-b85a-4926-a2f8-cd78c52915f6.png)

**Comment by andikrueger** [comment]
Good find. At the moment, the AZ.Resources is not part of the Dependencies. Will update accordingly.

The second error message was caused by a missing exclusion within an if statement.

**Comment by andikrueger** [comment]
As discussed in #2568 we are not adding the dependency to the core dependencies within M365DSC. 

The certificate issue got resolved by PR #2844
